spanner/api/Program.cs

```diff
@@ -0,0 +1,965 @@
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT AlbumId, AlbumTitle, MarketingBudget FROM Albums@ "
+                    + "{FORCE_INDEX=AlbumsByAlbumTitle} "
+                    + $"WHERE AlbumTitle >= '{startTitle}' "
+                    + $"AND AlbumTitle < '{endTitle}'");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]} "
+                        + $"MarketingBudget : {reader["MarketingBudget"]}");
+                    }
+                }
+            }
+            return;
+            // [END read_data_with_index]
+            // [END query_data_with_index]
+        }
+
+        public static async Task QueryDataWithStoringIndexAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START read_data_with_storing_index]
+            string startTitle = "Aardvark";
+            string endTitle = "Goo";
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Create connection to spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT AlbumId, AlbumTitle, MarketingBudget FROM Albums@ "
+                    + "{FORCE_INDEX=AlbumsByAlbumTitle2} "
+                    + $"WHERE AlbumTitle >= '{startTitle}' "
+                    + $"AND AlbumTitle < '{endTitle}'");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]} "
+                        + $"MarketingBudget : {reader["MarketingBudget"]}");
+                    }
+                }
+            }
+            return;
+            // [END read_data_with_storing_index]
+        }
+
+        public static async Task QueryDataWithTransactionAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START read_only_transaction]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/{instanceId}"
+            + $"/databases/{databaseId}";
+            // Gets a transaction object that captures the database state
+            // at a specific point in time.
+            using (TransactionScope scope = new TransactionScope(

```

Comment:
unfortunately, you will need to do another version of this for .net core -- where this type doesnt exist and you need to set .Transaction manually on each command.  (bummer, I know)